Fix When Different Layout

FurbyX92 opened the pull request:
This Change fix when layout change and dpi is not enought. When you change dpi to a value out of same layout (NORMAL to LARGE), you have to revert for install all apps on playstore.

furbyx-92 committed:
Fix When Different Layout

This Change fix when layout change and dpi is not enought. When you change dpi to a value out of same layout (NORMAL to LARGE), you have to revert for install all apps on playstore.

rovo89 commented:
Thanks. Any app and DPIs I could test this with?

Obviously, the screen size shouldn't be hardcoded, so I'm wondering whether it should be autocalculated or if it makes sense to give the user a choice to pick one.

FurbyX92 commented:
You can use ESPN SportsCenter, ESPN FC Soccer & World Cup or Microsoft Office Mobile. Only change DPI to Activate Phablet Mode and Large Layout, has to be like 240 to 280 if you are in 480dpi... I don't have idea in other smartphone, I have a galaxy s4 and with 280dpi is phablet mode, with 320dpi is LARGE layout. The apps only works with Phone Mode and Normal Layout.

FurbyX92 committed:
Another approach, now can be chosen the layout to send google.

FurbyX92 commented:
I made another pull, you can choose now the Layout with a ListPreferences

rovo89 commented:
Thanks! Changing the layout indeed seems to have an effect. I'll check if it possible to detect the best layout automatically based on the chosen DPI, or otherwise how the choice can be made easy for the user (I think most of them wouldn't know what to choose here).

FurbyX92 commented:
xlarge screens are at least 960dp x 720dp
large screens are at least 640dp x 480dp
normal screens are at least 470dp x 320dp
small screens are at least 426dp x 320dp

could be use the xdpi and ydpi; they are not modified by dpi modification.

http://developer.android.com/guide/practices/screens_support.html 
http://stefan222devel.blogspot.mx/2012/10/android-screen-densities-sizes.html

FurbyX92 committed:
Another approach, now can be chosen the layout to send google.

FurbyX92 committed:
Cleaning code, its more simple now.

FurbyX92 committed:
Adding String to Layout type in Layout List Selector.
Automatic detector is working, I think :D

ghost commented:
Hello everyone,

I greatly enjoy using the 'Play Store Fixes' Xposed module. Would it be possible to have an option in a future release of the 'Play Store Fixes' module to show the 'Play Store Fixes' application within the main 'app drawer' so that the module UI is able to be launched without the need for first launching the 'Xposed Installer' application? Thank you for your time and keep up the good work!

P.S. Sorry if this is not the correct place to ask this question, but I have never used github before. This account is simply one that is shared to anyone. Also, I tried first to see if I could contact rovo89 by email, but it appears that he doesn't have an email address that is available to the public.

FurbyX92 committed:
Modifications for clean code.
If auto calculate is selected then select layout is disabled.
Completed strings in strings.xml

FurbyX92 committed:
Change made for disable debug option.
Play Store 4.9.13 broke this feature.

FurbyX92 commented:
With the new update in PlayStore 4.9.13 the debug option was broken. I disable the preference and default value is false. I didn't delete the code because could be useful when found a solution. But in my opinion, debug option is useless.... at least I think that.
